### DISCUSSION TITLE
RQL where clause on first element of an array

### DISCUSSION THREAD

**Discussion by hshenMcc** [ravendb/ravendb#18839] [Q&A] [answered]
a document is defined as 

```
class TrackObject {
   List<TrackItem> TrackItems{get; set;}
  ... ... ...
}
class TrackItem {
       string CreatedBy {get; set;}
      ... ...
}
```

How can I make an RQL, so that it returns all TrackObject documents that the document's first TrackItem has CreatedBy == 'Joe"? I tried

  ```
 from TrackOject
  where TrackItems[0].CreatedBy == 'Joe'
```

It did not work. 

Thanks

**Comment by ayende** [comment]
There isn't a way to query based of position in an array using dynamic queries.
You have to use a static index for this.

**Reply by hshenMcc** [reply]
Thanks. Have no clue how to achieve this in a static index. Can you give a sample? Thanks again.

**Reply by ayende** [reply, marked as answer]
Define a static index with a `map`:

```
map("TrackOject", t =>{
 return {
    FirstItemCreatedBy = t.TrackItems[0].CreatedBy
 };
});
```

And then you query using:

```
from index 'MyIndex'
where FirstItemCreatedBy  = 'Joe'
```

**Reply by hshenMcc** [reply]
Thanks. Will try.
